### PR TITLE
formatting: Update code to standardize styling, remove linter

### PR DESCRIPTION
**Pull request by grandizzy**
- remove `ERC20` import from `ERC20PoolFactory.sol`
- used alignment where possible to improve readability (in both contracts and tests)
- standardize underscore syntax
`_internalVariable`
`publicVariable`
`memoryVariable`
`parameterVariable_`
`CONSTANT`
- improve tests readability
- organize imports

### CODE REVIEW

File: src/ERC20Pool.sol [hunk 1]
```diff
@@ -2,10 +2,12 @@
...
-import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import {BitMaps} from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
-import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-import {Clone} from "@clones/Clone.sol";
+import { ERC20 }        from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 }    from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { BitMaps } from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
+
+import { Clone } from "@clones/Clone.sol";
```
Comment (EdNoepel) [inline src/ERC20Pool.sol:5]: I'd expect whitespace padding to line up the `from` statements.

File: src/_test/tokens/Ajna.t.sol [hunk 2]
```diff
@@ -1,24 +1,24 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity 0.8.11;
 
-import {DSTestPlus} from "../utils/DSTestPlus.sol";
+import { AjnaToken }    from "../../tokens/Ajna.sol";
```
Comment (lucas-manuel) [inline src/_test/tokens/Ajna.t.sol]:
```solidity
import { AjnaToken } from "../../tokens/Ajna.sol";

import { DSTestPlus } from "../utils/DSTestPlus.sol";
```

File: src/_test/tokens/Ajna.t.sol [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+    function testFailCannotSendTokensToContract() external {
+        assert(false == _token.transfer(address(_token), 1));
     }
 
-    function invariantMetadata() public {
-        assertEq(token.name(), "Ajna");
-        assertEq(token.symbol(), "AJNA");
-        assertEq(token.decimals(), 18);
+    function invariantMetadata() external {
+        assertEq(_token.name(), "Ajna");
```
Comment (lucas-manuel) [inline src/_test/tokens/Ajna.t.sol]:
```solidity
assertEq(_token.name(),     "Ajna");
assertEq(_token.symbol(),   "AJNA");
assertEq(_token.decimals(), 18);
```

File: src/_test/utils/DSTestPlus.sol [hunk 4]
```diff
@@ -1,13 +1,48 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 pragma solidity 0.8.11;
 
-import {DSTest} from "@ds-test/test.sol";
-import {ERC20} from "@solmate/tokens/ERC20.sol";
-
-import {Test} from "@std/Test.sol";
-import {Vm} from "@std/Vm.sol";
+import { DSTest }   from "@ds-test/test.sol";
```
Comment (lucas-manuel) [inline src/_test/utils/DSTestPlus.sol]:
```solidity
import { DSTest } from "@ds-test/test.sol";
import { ERC20 }  from "@solmate/tokens/ERC20.sol";
import { Test }   from "@std/Test.sol";
import { Vm }     from "@std/Vm.sol";
```

File: src/_test/utils/DSTestPlus.sol [hunk 5]
```diff
@@ -1,13 +1,48 @@
...
+import { ERC20 }    from "@solmate/tokens/ERC20.sol";
+import { Test }     from "@std/Test.sol";
+import { Vm }   from "@std/Vm.sol";
 
 contract DSTestPlus is DSTest, Test {
+    // prices
+    uint256 internal _p50159        = 50_159.593888626183666006 * 1e18;
```
Comment (lucas-manuel) [inline src/_test/utils/DSTestPlus.sol]:
```solidity
    uint256 internal _p50159    = 50_159.593888626183666006 * 1e18;
    uint256 internal _p49910    = 49_910.043670274810022205 * 1e18;
    uint256 internal _p10016    = 10_016.501589292607751220 * 1e18;
    uint256 internal _p9020     = 9_020.461710444470171420 * 1e18;
    uint256 internal _p8002     = 8_002.824356287850613262 * 1e18;
    uint256 internal _p5007     = 5_007.644384905151472283 * 1e18;
    uint256 internal _p4000     = 4_000.927678580567537368 * 1e18;
    uint256 internal _p3514     = 3_514.334495390401848927 * 1e18;
    uint256 internal _p3010     = 3_010.892022197881557845 * 1e18;
    uint256 internal _p2850     = 2_850.155149230026939621 * 1e18;
    uint256 internal _p2835     = 2_835.975272865698470386 * 1e18;
    uint256 internal _p2821     = 2_821.865943149948749647 * 1e18;
    uint256 internal _p2807     = 2_807.826809104426639178 * 1e18;
    uint256 internal _p2793     = 2_793.857521496941952028 * 1e18;
    uint256 internal _p2779     = 2_779.957732832778084277 * 1e18;
    uint256 internal _p2503     = 2_503.519024294695168295 * 1e18;
    uint256 internal _p2000     = 2_000.221618840727700609 * 1e18;
    uint256 internal _p1004     = 1_004.989662429170775094 * 1e18;
    uint256 internal _p502      = 502.433988063349232760 * 1e18;
    uint256 internal _p146      = 146.575625611106531706 * 1e18;
    uint256 internal _p145      = 145.846393642892072537 * 1e18;
    uint256 internal _p100      = 100.332368143282009890 * 1e18;
    uint256 internal _p13_57    = 13.578453165083418466 * 1e18;
    uint256 internal _p5_26     = 5.263790124045347667 * 1e18;
    uint256 internal _p1_64     = 1.646668492116543299 * 1e18;
    uint256 internal _p1_31     = 1.315628874808846999 * 1e18;
    uint256 internal _p1_05     = 1.051140132040790557 * 1e18;
    uint256 internal _p0_951347 = 0.951347940696068854 * 1e18;
    uint256 internal _p0_607286 = 0.607286776171110946 * 1e18;
    uint256 internal _p0_189977 = 0.189977179263271283 * 1e18;
    uint256 internal _p0_006856 = 0.006856528811048429 * 1e18;
    uint256 internal _p0_006822 = 0.006822416727411372 * 1e18;
    uint256 internal _p0_000046 = 0.000046545370002462 * 1e18;
    uint256 internal _p1        = 1 * 1e18;
```

File: src/_test/utils/Tokens.sol [hunk 6]
```diff
@@ -1,13 +1,13 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity 0.8.11;
 
-import {ERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ERC20 }    from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
Comment (lucas-manuel) [inline src/_test/utils/Tokens.sol]: `import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";`

File: src/_test/utils/Users.sol [hunk 7]
```diff
@@ -1,53 +1,42 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity 0.8.11;
 
-import {IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import {ERC20Pool, IPool} from "../../ERC20Pool.sol";
+import { ERC20Pool }    from "../../ERC20Pool.sol";
```
Comment (lucas-manuel) [inline src/_test/utils/Users.sol]:
```solidity
import { ERC20Pool } from "../../ERC20Pool.sol";

import { IERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```

File: src/_test/BucketMathTest.t.sol [hunk 8]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity 0.8.11;
 
-import {DSTestPlus} from "./utils/DSTestPlus.sol";
+import { DSTestPlus }   from "./utils/DSTestPlus.sol";
```
Comment (lucas-manuel) [inline src/_test/BucketMathTest.t.sol]: Rm extra spaces

File: src/_test/BucketMathTest.t.sol [hunk 9]
```diff
@@ -25,75 +25,59 @@ contract BucketMathTest is DSTestPlus {
...
-        assertEq(BucketMath.priceToIndex(priceToTest), 4156);
+        assertEq(BucketMath.indexToPrice(4156),                 BucketMath.MAX_PRICE);
+        assertEq(BucketMath.priceToIndex(BucketMath.MAX_PRICE), 4156);
+        assertEq(BucketMath.indexToPrice(-3232),                BucketMath.MIN_PRICE);
+        assertEq(BucketMath.priceToIndex(BucketMath.MIN_PRICE), -3232);
 
-        priceToTest = 49_910.043670274810022205 * 10**18;
-        assertEq(BucketMath.indexToPrice(2169), priceToTest);
-        assertEq(BucketMath.priceToIndex(priceToTest), 2169);
+        assertEq(BucketMath.indexToPrice(2169),         _p49910);
```
Comment (lucas-manuel) [inline src/_test/BucketMathTest.t.sol]: Follow this format for all below assertions

```solidity
        assertEq(BucketMath.indexToPrice(2169),    _p49910);
        assertEq(BucketMath.priceToIndex(_p49910), 2169);
```

File: src/_test/ERC20Pool.t.sol [hunk 10]
```diff
@@ -1,47 +1,43 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity 0.8.11;
 
-import {ERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import {DSTestPlus} from "./utils/DSTestPlus.sol";
-import {UserWithCollateral, UserWithQuoteToken} from "./utils/Users.sol";
-import {CollateralToken, QuoteToken} from "./utils/Tokens.sol";
+import { CollateralToken, QuoteToken }              from "./utils/Tokens.sol";
```
Comment (lucas-manuel) [inline src/_test/ERC20Pool.t.sol]: Rm extra spaces

File: src/_test/ERC20Pool.t.sol [hunk 11]
```diff
@@ -1,47 +1,43 @@
...
+        _lender.approveToken(_quote, address(_pool), 200_000 * 1e18);
+        _borrower.approveToken(_collateral, address(_pool), 200_000 * 1e18);
     }
 
     // @notice:Tests pool factory inputs match the pool created
-    function testDeploy() public {
-        assertEq(address(collateral), address(pool.collateral()));
-        assertEq(address(quote), address(pool.quoteToken()));
+    function testDeploy() external {
+        assertEq(address(_collateral), address(_pool.collateral()));
```
Comment (lucas-manuel) [inline src/_test/ERC20Pool.t.sol:37]:
```solidity
        assertEq(address(_collateral), address(_pool.collateral()));
        assertEq(address(_quote),      address(_pool.quoteToken()));
```

File: src/_test/ERC20Pool.t.sol [hunk 12]
```diff
@@ -51,48 +47,48 @@ contract ERC20PoolTest is DSTestPlus {
...
     }
 
     // @notice: Check that initialize can only be called once
-    function testInitialize() public {
+    function testInitialize() external {
         uint256 initialInflator = 1 * 10**27;
 
-        assertEq(pool.inflatorSnapshot(), initialInflator);
-        assertEq(pool.lastInflatorSnapshotUpdate(), 0);
+        assertEq(_pool.inflatorSnapshot(),              initialInflator);
```
Comment (lucas-manuel) [inline src/_test/ERC20Pool.t.sol]: Rm extra spaces

File: src/_test/ERC20Pool.t.sol [hunk 13]
```diff
@@ -51,48 +47,48 @@ contract ERC20PoolTest is DSTestPlus {
...
         skip(8200);
 
-        borrower.addCollateral(pool, 2 * 1e18);
-        borrower.borrow(pool, 1000 * 1e18, 3000 * 1e18);
+        _borrower.addCollateral(_pool, 2 * 1e18);
+        _borrower.borrow(_pool, 1000 * 1e18, 3000 * 1e18);
 
-        assertGt(pool.inflatorSnapshot(), initialInflator);
-        assertEq(pool.lastInflatorSnapshotUpdate(), 8200);
+        assertGt(_pool.inflatorSnapshot(),              initialInflator);
```
Comment (lucas-manuel) [inline src/_test/ERC20Pool.t.sol]: Rm extra spaces

File: src/_test/ERC20Pool.t.sol [hunk 14]
```diff
@@ -51,48 +47,48 @@ contract ERC20PoolTest is DSTestPlus {
...
 
         // Attempt to call initialize() to reset global variables and check for revert
         vm.expectRevert(IPool.AlreadyInitialized.selector);
-        pool.initialize();
+        _pool.initialize();
 
         // check that global variables weren't reset
-        assertGt(pool.inflatorSnapshot(), initialInflator);
-        assertEq(pool.lastInflatorSnapshotUpdate(), 8200);
+        assertGt(_pool.inflatorSnapshot(),              initialInflator);
```
Comment (lucas-manuel) [inline src/_test/ERC20Pool.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 15]
```diff
@@ -1,269 +1,252 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity 0.8.11;
 
-import {DSTestPlus} from "./utils/DSTestPlus.sol";
-import {UserWithCollateral, UserWithQuoteToken} from "./utils/Users.sol";
-import {CollateralToken, QuoteToken} from "./utils/Tokens.sol";
+import { CollateralToken, QuoteToken }              from "./utils/Tokens.sol";
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 16]
```diff
@@ -1,269 +1,252 @@
...
-    UserWithQuoteToken internal lender;
-    UserWithCollateral internal bidder;
-
-    function setUp() public {
-        collateral = new CollateralToken();
-        quote = new QuoteToken();
-
-        ERC20PoolFactory factory = new ERC20PoolFactory();
-        pool = factory.deployPool(address(collateral), address(quote));
+import { Buckets }      from "../libraries/Buckets.sol";
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 17]
```diff
@@ -1,269 +1,252 @@
...
+        _bidder.purchaseBid(_pool, 4_000 * 1e18, _p4000);
 
         // check bidder and pool balances
-        assertEq(collateral.balanceOf(address(bidder)), 100 * 1e18);
-        assertEq(quote.balanceOf(address(bidder)), 0);
-        assertEq(collateral.balanceOf(address(pool)), 100 * 1e18);
-        assertEq(quote.balanceOf(address(pool)), 5_000 * 1e18);
-        assertEq(pool.totalQuoteToken(), 5_000 * 1e45);
-        assertEq(pool.totalCollateral(), 100 * 1e27);
+        assertEq(_collateral.balanceOf(address(_bidder)),   100 * 1e18);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 18]
```diff
@@ -1,269 +1,252 @@
...
+        assertEq(_pool.totalCollateral(),                   100 * 1e27);
 
         // check 4_000.927678580567537368 bucket balance before purchase bid
-        (, , , uint256 deposit, uint256 debt, , , uint256 bucketCollateral) = pool.bucketAt(
-            4_000.927678580567537368 * 1e18
-        );
-        assertEq(deposit, 0);
-        assertEq(debt, 3_000 * 1e45);
+        (, , , uint256 deposit, uint256 debt, , , uint256 bucketCollateral) = _pool.bucketAt(_p4000);
+        assertEq(deposit,           0);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 19]
```diff
@@ -1,269 +1,252 @@
...
+        assertEq(_pool.lup(),       _p1004);
         // check 4_000.927678580567537368 bucket balance after purchase bid
-        (, , , deposit, debt, , , bucketCollateral) = pool.bucketAt(
-            4_000.927678580567537368 * 1e18
-        );
-        assertEq(deposit, 0);
-        assertEq(debt, 1_000 * 1e45);
-        assertEq(bucketCollateral, 0.499884067064554306651186498 * 1e27);
+        (, , , deposit, debt, , , bucketCollateral) = _pool.bucketAt(_p4000);
+        assertEq(deposit,           0);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 20]
```diff
@@ -1,269 +1,252 @@
...
-        (, , , deposit, debt, , , bucketCollateral) = pool.bucketAt(
-            3_010.892022197881557845 * 1e18
-        );
-        assertEq(deposit, 0);
-        assertEq(debt, 3_000 * 1e45);
-        assertEq(bucketCollateral, 0);
+        (, , , deposit, debt, , , bucketCollateral) = _pool.bucketAt(_p3010);
+        assertEq(deposit,           0);
+        assertEq(debt,              3_000 * 1e45);
+        assertEq(bucketCollateral,  0);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 21]
```diff
@@ -1,269 +1,252 @@
...
-        (, , , deposit, debt, , , bucketCollateral) = pool.bucketAt(
-            1_004.989662429170775094 * 1e18
-        );
-        assertEq(deposit, 3_000 * 1e45);
-        assertEq(debt, 0);
-        assertEq(bucketCollateral, 0);
+        (, , , deposit, debt, , , bucketCollateral) = _pool.bucketAt(_p1004);
+        assertEq(deposit,           3_000 * 1e45);
+        assertEq(debt,              0);
+        assertEq(bucketCollateral,  0);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 22]
```diff
@@ -1,269 +1,252 @@
...
 
         // check bidder and pool balances
-        assertEq(collateral.balanceOf(address(bidder)), 99.500115932935445694 * 1e18);
-        assertEq(quote.balanceOf(address(bidder)), 2_000 * 1e18);
-        assertEq(collateral.balanceOf(address(pool)), 100.499884067064554306 * 1e18);
-        assertEq(quote.balanceOf(address(pool)), 3_000 * 1e18);
-        assertEq(pool.totalQuoteToken(), 3_000 * 1e45);
-        assertEq(pool.totalCollateral(), 100 * 1e27);
+        assertEq(_collateral.balanceOf(address(_bidder)),   99.500115932935445694 * 1e18);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 23]
```diff
@@ -1,269 +1,252 @@
...
 
         // check bidder and pool balances
-        assertEq(collateral.balanceOf(address(bidder)), 100 * 1e18);
-        assertEq(quote.balanceOf(address(bidder)), 0);
-        assertEq(collateral.balanceOf(address(pool)), 100 * 1e18);
-        assertEq(quote.balanceOf(address(pool)), 5_000 * 1e18);
-        assertEq(pool.totalCollateral(), 100 * 1e27);
-        assertEq(pool.hpb(), p4000);
-        assertEq(pool.lup(), p3010);
+        assertEq(_collateral.balanceOf(address(_bidder)),   100 * 1e18);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

File: src/_test/ERC20PoolBid.t.sol [hunk 24]
```diff
@@ -1,269 +1,252 @@
...
+        assertEq(_pool.hpb(),                               _p4000);
+        assertEq(_pool.lup(),                               _p3010);
 
         // check 4_000.927678580567537368 bucket balance before purchase Bid
-        (, , , uint256 deposit, uint256 debt, , , uint256 bucketCollateral) = pool.bucketAt(p4000);
-        assertEq(deposit, 0);
-        assertEq(debt, 1_000 * 1e45);
-        assertEq(bucketCollateral, 0);
+        (, , , uint256 deposit, uint256 debt, , , uint256 bucketCollateral) = _pool.bucketAt(_p4000);
+        assertEq(deposit,           0);
```
Comment (lucas-manuel) [inline src/_test/ERC20PoolBid.t.sol]: Rm extra spaces

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Comment by grandizzy** [comment]
@lucas-manuel I addressed rm extra spaces with https://github.com/ajna-finance/contracts/pull/99/commits/4fa9531694f8f41da4973734555ff8fc72aa66e1 pls recheck

File: src/libraries/Buckets.sol [hunk 25]
```diff
@@ -23,176 +23,177 @@ library Buckets {
...
-                    curLup.debt = 0;
+                    curLup.debt      = 0;
                 } else {
                     // pay as much debt as possible and exit
-                    curLup.onDeposit += _amount;
-                    curLup.debt -= _amount;
-                    _amount = 0;
+                    curLup.onDeposit += amount_;
+                    curLup.debt      -= amount_;
+                    amount_         = 0;
```
Comment (EdNoepel) [inline src/libraries/Buckets.sol]: Whitespace on lines 191 and 196 was altered, but the operators still don't line up.

**Review by EdNoepel — APPROVED** [review]
I prefer we merge this sooner, rather than amassing merge conflicts.  Significant improvement in readability.

File: src/libraries/Buckets.sol (hunk 25 above)
Comment (grandizzy) [reply src/libraries/Buckets.sol]: Fixed with https://github.com/ajna-finance/contracts/pull/99/commits/fc50f3d58fa836cfae024a4155b23fc8978b9e96

File: src/ERC20Pool.sol (hunk 1 above)
Comment (grandizzy) [reply src/ERC20Pool.sol:5]: yeah, the padding applies only in scope of the same group, I split here in two groups as they are different functionality @lucas-manuel you think that's OK or they should still be same group?

**Review by MikeHathaway — APPROVED** [review]
LGTM!

**Pull request merged**